### ISSUE TITLE
Custom back-button widget

### ISSUE THREAD

**Issue by hanaral** [elementary/granite#484]
## Problem

The current solution isn't a long term one, and it likely going to look weird to new users who don't see icons inside buttons anywhere else.

## Proposal

Now that libadwaita is a thing, I don't see why Granite shouldn't be able to actually function like a custom library with its own custom widgets. Since flatpak allows apps to bring the elementary stylesheet with them on other platforms the way they look shouldn't be too much of an issue anyway.

## Prior Art

A quick mock-up:

![granite-back-button](https://user-images.githubusercontent.com/58219504/114687127-54edbe00-9d0b-11eb-8c64-71a0739648ae.png)

**Comment by cassidyjames** [comment]
There was never anything stopping this technically, someone just needs to write the widget and ensure it picks up style and everything correctly. I do like the shaped button more than the icon button. 🙂

**Comment by hanaral** [comment]
> There was never anything stopping this technically, someone just needs to write the widget and ensure it picks up style and everything correctly. I do like the shaped button more than the icon button. slightly_smiling_face

Just to confirm, were talking about the same thing (custom widget drawing) here, right?
Maybe someone could kick this off by looking at how libhandy/adwaita does it, since they have plenty of custom stuff!

**Comment by cassidyjames** [comment]
>Just to confirm, were talking about the same thing (custom widget drawing) here, right?
>Maybe someone could kick this off by looking at how libhandy/adwaita does it, since they have plenty of custom stuff!

Yes. I think the issue is knowing how the border and shadows should work with an angled button, but I'm not personally familiar with how custom drawing works.
